src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
+@SuppressWarnings("unchecked")
 public class ArrayList<T> implements List<T> {
+    static final int INITIAL_SIZE = 10;
+    static final double growIndex = 1.5;
```

Comment:
naming convention: https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names
```suggestion
    static final double GROW_INDEX = 1.5;
```